synthesizer/snark/src/proving_key/mod.rs

```diff
@@ -20,7 +20,7 @@ mod serialize;
 
 use std::collections::BTreeMap;
 
-#[derive(Clone)]
+#[derive(Clone, Eq)]
```

Comment:
I'd recommend adding:
```
impl<N: Network> Eq for ProvingKey<N> {}
```
next to `PartialEq` for better readability. Otherwise the reader may see only the `PartialEq` impl and not realize `Eq` is also impl'ed.

Reply:
[bc9fc63](https://github.com/AleoHQ/snarkVM/pull/2032/commits/bc9fc63321c1ce3b8649c314043657a7c2f3cb44)